cmd/minishift/cmd/oc_env.go

```diff
@@ -42,12 +44,19 @@ type OcShellConfig struct {
 }
 
 func getOcShellConfig(ocPath, forcedShell string) (*OcShellConfig, error) {
+	var cmdLine string
+
 	userShell, err := shell.GetShell(forcedShell)
 	if err != nil {
 		return nil, err
 	}
 
-	cmdLine := "minishift oc-env"
+	if constants.IsProfileExists() {
+		cmdLine = "minishift oc-env --profile " + constants.GetMachineName()
```

Comment:
@gbraad Also brought up the `default` profile point. I think it depends on how we want to shape the user experience and changing this would change the directory structure when no profile is used as at this point by default the instance name is `minishift`.